packages/logging/lokiLogProvider.go

```diff
@@ -98,11 +92,94 @@ func (llp *LokiLogProvider) getLogs(doguName string, linesCount int) ([]logLine,
 		result = result[:linesCount]
 	}
 
-	logrus.Debugf("finished loki query; got %d logLines", len(result))
+	logrus.Debugf("finished loading logs for linecount; got %d logLines", len(result))
+
+	return result, nil
+}
+
+// queryLogs queries logs from loki for the given time-window (startDate and endDate).
+// The allowed time-window is max 30 days (limited by loki).
+// Since loki also has a limit of max 5000 log-lines per request, a query can result in multiple request for the loki backend.
+func (llp *LokiLogProvider) queryLogs(doguName string, startDate *time.Time, endDate *time.Time, filter string) ([]logLine, error) {
```

Comment:
I think there is no need to pass times as pointers. To check for Zero Values there is a function IsZero in the time pkg.

Reply:
:+1: